App/Logic/AnalyticsLogic.swift

```diff
@@ -199,11 +300,76 @@ extension Logic.Analytics {
   }
 
   /// Updates the opportunity window for the event without exposure
-  struct UpdateEventWithoutExposureOppurtunityWindow: AppStateUpdater {
+  struct SetEventWithoutExposureOpportunityWindow: AppStateUpdater {
     let window: AnalyticsState.OpportunityWindow
 
     func updateState(_ state: inout AppState) {
       state.analytics.eventWithoutExposureWindow = self.window
     }
   }
+
+  /// Sets the opportunity window for the dummy analytics traffic
+  struct SetDummyTrafficOpportunityWindow: AppStateUpdater {
+    let dummyTrafficStochasticDelay: Double
+    let now: Date
+
+    func updateState(_ state: inout AppState) {
+      let windowStart = self.now.addingTimeInterval(self.dummyTrafficStochasticDelay)
+      let windowDuration = AnalyticsState.OpportunityWindow.secondsInDay
+      state.analytics.dummyTrafficOpportunityWindow = .init(windowStart: windowStart, windowDuration: windowDuration)
+    }
+  }
+}
+
+// MARK: Helpers
+
+private extension AnalyticsRequest.Body {
+  /// Creates a dummy request
+  static func dummy(deviceTokenLength: Int) -> Self {
+    return Self(
+      province: Province.allCases.randomElement() ?? AppLogger.fatalError("No provinces defined"),
+      exposureNotificationStatus: ExposureNotificationStatus.randomCase(),
+      pushNotificationStatus: UNAuthorizationStatus.randomCase(),
+      riskyExposureDetected: Bool.random(),
+      deviceToken: String.random(length: deviceTokenLength)
```

Comment:
Isn't it better to generate a random bunch of data and then encode it as base64?

Reply:
Actually the correct device token is required. I changed the implementation.